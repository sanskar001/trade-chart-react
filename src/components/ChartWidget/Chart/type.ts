import { Time } from "lightweight-charts";

export interface CandleDataType {
  time: Time;
  open: number;
  close: number;
  high: number;
  low: number;
}

export type CandleSeriesDataType = Array<CandleDataType>;
