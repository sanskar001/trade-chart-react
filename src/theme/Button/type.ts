type ButtonType = "button" | "submit" | "reset";

type ButtonClassType = "btn-outlined" | "";

type ClickHandlerType = React.MouseEventHandler<HTMLButtonElement>;

type PropsType =
  | string
  | boolean
  | undefined
  | ButtonType
  | React.ReactNode
  | ClickHandlerType
  | ButtonClassType
  | React.CSSProperties;

export interface ButtonProps {
  children: React.ReactNode;
  type?: ButtonType;
  onClick?: ClickHandlerType;
  btnClass?: ButtonClassType;
  className?: string;
  style?: React.CSSProperties;
  disabled?: boolean;
  [key: string]: PropsType;
}
