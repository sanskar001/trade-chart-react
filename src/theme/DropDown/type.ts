import React from "react";

interface OptionType<T> {
  value: T;
  label: React.ReactNode;
  shortLabel?: React.ReactNode;
  isFrequent?: boolean;
}

export type OptionListType<T> = Array<OptionType<T>>;

export interface DropDownProps {
  optionList: OptionListType<string>;
  selectedValue: string;
  onSelect: (_value: string) => void;
  className?: string;
  style?: React.CSSProperties;
}
