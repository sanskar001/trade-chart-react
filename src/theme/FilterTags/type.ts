export interface Option {
  value: string;
  label: React.ReactNode;
}

type SetFilterType = React.Dispatch<React.SetStateAction<string>>;

export interface FilterTagsProps {
  filterOptions: Option[];
  selectedValue: string;
  setFilter: SetFilterType;
  name?: string;
  className?: string;
  style?: React.CSSProperties;
}
