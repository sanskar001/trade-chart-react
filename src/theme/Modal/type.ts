type CloseHandlerType = () => void;

export interface BackDropProps {
  backdropClass?: string;
  onClose: CloseHandlerType;
}

export interface ModalWindowProps {
  children: React.ReactNode;
  onClose: CloseHandlerType;
  heading?: React.ReactNode;
  modalClass?: string;
  footerText?: string;
}

export interface ModalProps extends BackDropProps, ModalWindowProps {
  isShowModal: boolean;
}
