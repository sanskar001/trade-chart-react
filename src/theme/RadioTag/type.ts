type ChangeHandlerType = React.ChangeEventHandler<HTMLInputElement>;

export interface RadioTagProps {
  value: string;
  name: string;
  checked?: boolean;
  onChange?: ChangeHandlerType;
  label?: React.ReactNode;
  className?: string;
  style?: React.CSSProperties;
  disabled?: boolean;
}
