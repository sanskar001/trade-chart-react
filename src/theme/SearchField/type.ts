type ChangeHandlerType = React.ChangeEventHandler<HTMLInputElement>;

type PropsType =
  | string
  | boolean
  | undefined
  | ChangeHandlerType
  | React.CSSProperties;

export interface SearchFieldProps {
  value: string;
  onChange: ChangeHandlerType;
  className?: string;
  style?: React.CSSProperties;
  autoFocus?: boolean;
  disabled?: boolean;
  placeholder?: string;
  [key: string]: PropsType;
}
