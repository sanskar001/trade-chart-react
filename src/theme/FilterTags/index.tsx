import React from "react";
import { formatClass } from "../../util/formatClass";
import { FilterTagsProps } from "./type";

const FilterTags: React.FC<FilterTagsProps> = ({
  filterOptions,
  selectedValue,
  setFilter,
  name = "filter",
  className = "",
  style,
}) => {
  return (
    <div className={formatClass(`filter-tags ${className}`)} style={style}>
      {filterOptions.map((item) => {
        return (
          <div className="filter-tag" key={item.value}>
            <input
              type="radio"
              id={item.value}
              name={name}
              value={item.value}
              checked={item.value === selectedValue}
              onChange={setFilter.bind(null, item.value)}
            />
            <label htmlFor={item.value}>{item.label || item.value}</label>
          </div>
        );
      })}
    </div>
  );
};

export default FilterTags;
