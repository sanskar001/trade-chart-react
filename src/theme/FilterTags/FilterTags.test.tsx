import renderer from "react-test-renderer";
import FilterTags from ".";
import { FilterOption } from "./type";

const filterOptions: FilterOption[] = [
  { value: "first", label: "First Tag" },
  { value: "second", label: "Second Tag" },
];

const setFilter = jest.fn();

it("renders correctly when there is default filterTags", () => {
  const tree = renderer
    .create(
      <FilterTags
        filterOptions={filterOptions}
        selectedValue="first"
        setFilter={setFilter}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is filterTags with custom class", () => {
  const tree = renderer
    .create(
      <FilterTags
        filterOptions={filterOptions}
        selectedValue="first"
        setFilter={setFilter}
        className="p-4 border"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is filterTags with custom style", () => {
  const tree = renderer
    .create(
      <FilterTags
        filterOptions={filterOptions}
        selectedValue="first"
        setFilter={setFilter}
        style={{ padding: "4px" }}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders correctly when there is filterTags with custom name", () => {
  const tree = renderer
    .create(
      <FilterTags
        filterOptions={filterOptions}
        selectedValue="first"
        setFilter={setFilter}
        style={{ padding: "4px" }}
        name="symbol filter"
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
