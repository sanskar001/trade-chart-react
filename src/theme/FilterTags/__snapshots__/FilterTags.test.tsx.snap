// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is default filterTags 1`] = `
<div
  className="filter-tags"
>
  <div
    className="filter-tag"
  >
    <input
      checked={true}
      id="first"
      name="filter"
      onChange={[Function]}
      type="radio"
      value="first"
    />
    <label
      htmlFor="first"
    >
      First Tag
    </label>
  </div>
  <div
    className="filter-tag"
  >
    <input
      checked={false}
      id="second"
      name="filter"
      onChange={[Function]}
      type="radio"
      value="second"
    />
    <label
      htmlFor="second"
    >
      Second Tag
    </label>
  </div>
</div>
`;

exports[`renders correctly when there is filterTags with custom class 1`] = `
<div
  className="filter-tags p-4 border"
>
  <div
    className="filter-tag"
  >
    <input
      checked={true}
      id="first"
      name="filter"
      onChange={[Function]}
      type="radio"
      value="first"
    />
    <label
      htmlFor="first"
    >
      First Tag
    </label>
  </div>
  <div
    className="filter-tag"
  >
    <input
      checked={false}
      id="second"
      name="filter"
      onChange={[Function]}
      type="radio"
      value="second"
    />
    <label
      htmlFor="second"
    >
      Second Tag
    </label>
  </div>
</div>
`;

exports[`renders correctly when there is filterTags with custom name 1`] = `
<div
  className="filter-tags"
  style={
    Object {
      "padding": "4px",
    }
  }
>
  <div
    className="filter-tag"
  >
    <input
      checked={true}
      id="first"
      name="symbol filter"
      onChange={[Function]}
      type="radio"
      value="first"
    />
    <label
      htmlFor="first"
    >
      First Tag
    </label>
  </div>
  <div
    className="filter-tag"
  >
    <input
      checked={false}
      id="second"
      name="symbol filter"
      onChange={[Function]}
      type="radio"
      value="second"
    />
    <label
      htmlFor="second"
    >
      Second Tag
    </label>
  </div>
</div>
`;

exports[`renders correctly when there is filterTags with custom style 1`] = `
<div
  className="filter-tags"
  style={
    Object {
      "padding": "4px",
    }
  }
>
  <div
    className="filter-tag"
  >
    <input
      checked={true}
      id="first"
      name="filter"
      onChange={[Function]}
      type="radio"
      value="first"
    />
    <label
      htmlFor="first"
    >
      First Tag
    </label>
  </div>
  <div
    className="filter-tag"
  >
    <input
      checked={false}
      id="second"
      name="filter"
      onChange={[Function]}
      type="radio"
      value="second"
    />
    <label
      htmlFor="second"
    >
      Second Tag
    </label>
  </div>
</div>
`;
